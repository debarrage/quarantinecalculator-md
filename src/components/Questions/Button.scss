.buttons__button {
    text-align: center;
}

