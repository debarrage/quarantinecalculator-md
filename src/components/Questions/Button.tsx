import React from "react";
import { useDispatch } from "react-redux";
import { previousQuestionAction } from "../../store/actions";

export const Generic: React.FC<{ handler: () => void, name?: string}> = (props) => {
    const className = props.name ? `buttons_${props.name}` : "";
    return (
        <div className={`buttons_button ${className}`} onClick={props.handler}>
            {props.children}
        </div>
    );
};

export const Previous: React.FC = () => {

    const dispatch = useDispatch();
    const previous = () => {
        dispatch(previousQuestionAction());
    }

    return (<Generic handler={previous} name="previous">Vorige</Generic>);
};
