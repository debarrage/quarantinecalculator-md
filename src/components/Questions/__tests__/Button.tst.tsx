import React from "react";
import renderer from "react-test-renderer";
import * as Button from "../Button";

describe("Button classes", () => { 
    it("Should render generic button", () => {
        // Act
        const tree = renderer.create(<Button.Generic name="test" handler={() => {}} />).toJSON();
    
        // Assert
        expect(tree).toMatchSnapshot();
    });
});
