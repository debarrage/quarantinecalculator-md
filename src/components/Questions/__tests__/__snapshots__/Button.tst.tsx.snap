// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button classes Should render generic button 1`] = `
<div
  className="buttons__button buttons__test"
  onClick={[Function]}
/>
`;
